python/ray/data/_internal/bulk_dataset_iterator.py

```diff
@@ -0,0 +1,117 @@
+from typing import TYPE_CHECKING, Dict, List, Optional, Union, Iterator
+import warnings
+
+from ray.data.block import DataBatch
+from ray.data.dataset_iterator import DatasetIterator
+
+if TYPE_CHECKING:
+    import tensorflow as tf
+    import torch
+    from ray.data import Dataset
+    from ray.data._internal.torch_iterable_dataset import TorchTensorBatchType
+
+
+class BulkDatasetIterator(DatasetIterator):
+    def __init__(
+        self,
+        base_dataset: "Dataset",
+    ):
+        self._base_dataset = base_dataset
```

Comment:
Supernit: Could probably just call this `self._dataset` and the arg `dataset`, since we're not creating additional derived datasets as the `base_` implies.